pyccel/parser/semantic.py

```diff
@@ -993,7 +995,8 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
 
                 # ...
                 # Add memory allocation if needed
-                if lhs.allocatable:
+                array_declared_in_function = isinstance(rhs, FunctionCall) and not isinstance(rhs.funcdef, PyccelFunctionDef) and not rhs.funcdef.is_elemental and not isinstance(lhs, HomogeneousTupleVariable)
+                if lhs.allocatable and not array_declared_in_function and not isinstance(rhs, PyccelOperator):
```

Comment:
Why do you need ` and not isinstance(rhs, PyccelOperator)` here ? That means you don't want to allocate `c` when you do:
```python
def f(a : 'int[:]', b : 'int[:]'):
    c = a+b
```
Doesn't it?
Isn't this what is causing all the seg faults in the tests?

Reply:
to prevent creating an allocate node for `single_return` in `a = single_return() + 1`